app/modules/autogenerated_names/models.py

```diff
@@ -0,0 +1,60 @@
+# -*- coding: utf-8 -*-
+"""
+Autogeneratednames database models
+--------------------
+"""
+
+import enum
+import uuid
+
+from app.extensions import HoustonModel, db
+
+AUTOGEN_NAME_PREFIX_MIN_LENGTH = 2
+AUTOGEN_NAME_PREFIX_MAX_LENGTH = 10
+
+
+class AutogeneratedNameType(str, enum.Enum):
+    species = 'species'
+    region = 'region'
+    project = 'project'
+    organization = 'organization'
+
+
+class AutogeneratedName(db.Model, HoustonModel):
+    """
+    AutogeneratedNames database model.
+    """
+
+    guid = db.Column(
+        db.GUID, default=uuid.uuid4, primary_key=True
+    )  # pylint: disable=invalid-name
+
+    type = db.Column(
+        db.Enum(AutogeneratedNameType),
+        default=AutogeneratedNameType.species,
+        nullable=False,
+        index=True,
+    )
+    prefix = db.Column(db.String(), index=True, nullable=False)
+    # taxonomy guid for species
+    reference_guid = db.Column(db.GUID, index=True, default=uuid.uuid4)
+    next_value = db.Column(db.Integer, default=0)
+
+    def __repr__(self):
+        return (
+            '<{class_name}('
+            'guid={self.guid}, '
+            'type={self.type}, '
+            'prefix={self.prefix}, '
+            'reference_guid={self.reference_guid}'
+            ')>'.format(class_name=self.__class__.__name__, self=self)
+        )
+
+    def get_next(self):
+        next_val = str(self.next_value).zfill(4)
+
+        with db.session.begin():
+            # Update and write as close as possible to minimise race condition seen in old Wildbook
+            self.next_value += 1
```

Comment:
tight enough for you @naknomum ?

Reply:
i think it works -- for now.  :grin:   more importantly, you have a nice little method we can change if it proves not to be.  :heavy_check_mark: